### PR TITLE
Add missing stubs for multiprocessing.connection module.

### PR DESCRIPTION
**Pull request by devxpy**
Also ran it through the black code formatter, if you don't mind :)

[🐍🏕️](http://www.pycampers.com/)

### CODE REVIEW

**Comment by srittau** [comment]
Thank you for the pull request. Using black is actually counter-productive, though, since we use a different style guide for stubs, focusing on an "index" look and feel. See the `CONTRIBUTING` file for details.

**Comment by devxpy** [comment]
Please suggest how to solve this:

```python
...
class Listener:
    ...
    def __enter__(self) -> Listener: ...
...
```

Notice how `__enter__` returns a `Listener`.

`flake8` says-

```
$ flake8 stdlib/3/multiprocessing/connection.pyi 
stdlib/3/multiprocessing/connection.pyi:54:28: F821 undefined name 'Listener'
```

So if I add an import, flake8 is silenced.
```python
from multiprocessing.connection import Listener
```

`mypy` says - 
```
$ mypy stdlib/3/multiprocessing/connection.pyi 
stdlib/3/multiprocessing/connection.pyi:41: error: Name 'Listener' already defined (possibly by an import)
```

**Comment by JelleZijlstra** [comment]
You should ignore flake8 here. Stub files do not obey normal Python rules, and not all lint rules for normal Python apply.

**Comment by JelleZijlstra** [comment]
Also, @srittau black has a special mode for pyi files, which is compatible with our conventions.

**Comment by srittau** [comment]
True, I forgot about black's pyi mode, although it needs to be run with `-l130` (ambv/black#341) and there is another inconsistency (#2598).

File: stdlib/3/multiprocessing/connection.pyi [hunk 1]
```diff
@@ -1,36 +1,61 @@
-from typing import Any, Iterable, List, Optional, Tuple, Type, Union
 import socket
 import types
+from multiprocessing.connection import Listener
```
Comment (srittau) [inline stdlib/3/multiprocessing/connection.pyi]: This line is what causes CI to fail. We are in the `multiprocessing.connection` module, after all.

File: stdlib/3/multiprocessing/connection.pyi [hunk 2]
```diff
@@ -1,36 +1,61 @@
...
 # https://docs.python.org/3/library/multiprocessing.html#address-formats
 _Address = Union[str, Tuple[str, int]]
 
+class _ConnectionBase:
+    @property
+    def closed(self) -> bool: ...
+    @property
+    def readable(self) -> bool: ...
+    @property
+    def writable(self) -> bool: ...
```
Comment (srittau) [inline stdlib/3/multiprocessing/connection.pyi]: Please add `# undocumented` comments to these properties. Alternatively (and probably preferably) you could file a bug report against Python's documentation in the [bug tracker](https://bugs.python.org/).

File: stdlib/3/multiprocessing/connection.pyi [hunk 3]
```diff
@@ -1,36 +1,61 @@
...
+        self, buf: bytes, offset: int = ..., size: Optional[int] = ...
+    ) -> None: ...
+    def send(self, obj: Any) -> None: ...
+    def recv_bytes(self, maxlength: Optional[int] = ...) -> bytes: ...
+    def recv_bytes_into(self, buf: Any, offset: int = ...) -> int: ...
+    def recv(self) -> Any: ...
+    def poll(self, timeout: Optional[float] = ...) -> bool: ...
+
+class Connection(_ConnectionBase): ...
+class PipeConnection(_ConnectionBase): ...
```
Comment (srittau) [inline stdlib/3/multiprocessing/connection.pyi]: You tried to use `if _winapi:` here. While that does not work, `if sys.platform == "win32":` should.

**Comment by devxpy** [comment]
@srittau Thanks, I had removed the black formatter altogether, since it doesn't follow the style guide.
Will do the changes you suggested and push asap

File: stdlib/3/multiprocessing/connection.pyi (hunk 2 above)
Comment (srittau) [reply stdlib/3/multiprocessing/connection.pyi]: Also, the `closed` property.

**Pull request merged**